internal/core/model/adapters.rs

```diff
@@ -413,3 +415,511 @@ fn test_filter_model() {
+                    &self.wrapped_model.borrow().row_data(row).unwrap(),
+                ) == core::cmp::Ordering::Less
+            })
+            .map(|e| e.0)
+            .collect::<Vec<usize>>()
+            .first()
+        {
+            *insertion_index
+        } else {
+            self.mapping.borrow().len()
+        };
+
+        self.mapping.borrow_mut().insert(insertion_index, row);
+
+        if insertion_index == removed_index {
+            self.notify.row_changed(removed_index);
+        } else {
+            self.notify.row_removed(removed_index, 1);
+            self.notify.row_added(insertion_index, 1);
+        }
+    }
+
+    fn row_added(&self, index: usize, count: usize) {
+        if count == 0 {
+            return;
+        }
+
+        // Adjust the existing sorted row indices to match the updated source model
+        for row in self.mapping.borrow_mut().iter_mut() {
+            if *row >= index {
+                *row += count;
+            }
+        }
+
+        for row in index..(index + count) {
+            let insertion_index = if let Some(insertion_index) = self
+                .mapping
+                .borrow()
+                .iter()
+                .enumerate()
+                .skip_while(|e| {
+                    self.sort_helper.sort(
+                        // inserted value
+                        &self.wrapped_model.borrow().row_data(*e.1).unwrap(),
+                        &self.wrapped_model.borrow().row_data(row).unwrap(),
+                    ) == core::cmp::Ordering::Less
+                })
+                .map(|e| e.0)
+                .collect::<Vec<usize>>()
+                .first()
+            {
+                *insertion_index
+            } else {
+                self.mapping.borrow().len()
+            };
+
+            self.mapping.borrow_mut().insert(insertion_index, row);
+            self.notify.row_added(insertion_index, 1)
+        }
+    }
+
+    fn row_removed(&self, index: usize, count: usize) {
+        if count == 0 {
+            return;
+        }
+
+        let mut removed_rows = vec![];
+
+        let mut i = 0;
+
+        loop {
+            if i >= self.mapping.borrow().len() {
+                break;
+            }
+
+            let sort_index = *self.mapping.borrow().get(i).unwrap();
+
+            if sort_index >= index {
+                if sort_index < index + count {
+                    removed_rows.push(i);
+                    self.mapping.borrow_mut().remove(i);
+                    continue;
+                } else {
+                    *self.mapping.borrow_mut().get_mut(i).unwrap() -= count;
+                }
+            }
+
+            i += 1;
+        }
+
+        for removed_row in removed_rows {
+            self.notify.row_removed(removed_row, 1);
+        }
+    }
+
+    fn reset(&self) {
+        self.build_mapping_vec();
+        self.notify.reset();
+    }
+}
+
+/// Provides a sorted subset of rows by another [`Model`].
```

Comment:
I think the sort model does not limit the view, so it's not a subset :)

```suggestion
/// Provides a sorted view of rows by another [`Model`].
```